tornado-api/src/main/java/uk/ac/manchester/tornado/api/types/arrays/ByteArray.java

```diff
@@ -190,6 +190,16 @@ public int getSize() {
      */
     @Override
     public MemorySegment getSegment() {
+        return segment.asSlice(TornadoNativeArray.ARRAY_HEADER);
+    }
+
+    /**
+     * Returns the underlying {@link MemorySegment} of the {@code ByteArray} instance, including the header.
```

Comment:
Substitute all `{@code XXX}` with `{@link XXX}` in the Java doc for the new functions.

Reply:
these were introduced in the original PR, but i ll fix them